api/test/e2e/route_with_priority_test.go

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package e2e
+
+import (
+	"net/http"
+	"testing"
+)
+
+func TestRoute_with_priority(t *testing.T) {
+	tests := []HttpTestCase{
+		{
+			caseDesc: "add route with priority (-1)",
+			Object:   MangerApiExpect(t),
+			Method:   http.MethodPut,
+			Path:     "/apisix/admin/routes/r1",
+			Body: `{
+					  "uri": "/server_port",
+					  "methods": ["GET"],
+					  "priority": -1,
```

Comment:
do we support -1 as priority？

Reply:
yes, I have tested the priority of -2, -1, etc. and found that the priority supports negative values and conforms to the documentation, ` Larger value means higher priority. `.

Reply:
ping @moonming

Reply:
I think negative numbers are meaningless, maybe https://github.com/api7/lua-resty-radixtree should not allow negative numbers as a priority

Reply:
ok, I removed case which priority is a negative number. It is a `hidden attribute`.